src/functions/bathymetry.ts

```diff
@@ -14,33 +16,36 @@ import project from "../../project";
 
 // @ts-ignore
 import geoblaze, { Georaster } from "geoblaze";
-import { clipSketchToGeography } from "../util/clipSketchToGeography";
-import { BathymetryResults, ExtraParams } from "../types";
-import { getParamStringArray } from "../util/extraParams";
+import { clipToGeography } from "../util/clipToGeography";
+import { BathymetryResults, DefaultExtraParams } from "../types";
 
 export async function bathymetry(
-  sketch: Sketch<Polygon> | SketchCollection<Polygon>,
-  extraParams?: ExtraParams
+  sketch:
+    | Sketch<Polygon | MultiPolygon>
+    | SketchCollection<Polygon | MultiPolygon>,
+  extraParams: DefaultExtraParams = {}
 ): Promise<BathymetryResults> {
-  const geographyId = extraParams
-    ? getParamStringArray("geographyIds", extraParams)[0]
-    : undefined;
-  const clippedSketch = await clipSketchToGeography(sketch, geographyId);
+  const geographyId = getFirstFromParam("geographyIds", extraParams);
+  const curGeography = project.getGeographyById(geographyId, {
+    fallbackGroup: "default-boundary",
+  });
+
+  const clippedSketch = await clipToGeography(sketch, curGeography);
   const mg = project.getMetricGroup("bathymetry");
   const sketches = toSketchArray(clippedSketch);
   const box = clippedSketch.bbox || bbox(clippedSketch);
   if (!mg.classes[0].datasourceId)
     throw new Error(`Expected datasourceId for ${mg.classes[0]}`);
   const url = `${project.dataBucketUrl()}${getCogFilename(
-    mg.classes[0].datasourceId
+    project.getInternalRasterDatasourceById(mg.classes[0].datasourceId)
```

Comment:
Notice there's a couple new helper functions for fine-grained requests with tighter return types. e.g. `project.getInternalRasterDatasourceById`